virttest/graphical_console.py

```diff
@@ -229,6 +409,90 @@ def _key_event(self, down, keys):
             events.append(event)
         self._vm.qmp_monitors[0].input_send_event(events)
 
+    def _btn_event(self, event, btn):
+        """
+        Send mouse button event.
+
+        :param event: Button event, defined as True=down, False=up.
+        :param btn: Mouse button.
+        """
+
+        events = [{"type": "btn",
+                   "data": {"down": event,
+                            "button": btn}}]
+        self._vm.monitor.input_send_event(events)
+
+    def _scroll_event(self, event):
+        """
+        Send mouse scroll event.
+
+        :param event: Event as vertical forward/backward or horizontal left/right.
+                      But qmp only support scroll vertical forward/backward.
+                      Does not support scroll horizontal left/right currently.
+        """
+
+        self._btn_event(self.BTN_DOWN, event)
```

Comment:
typo?

Reply:
For qmp_console, send btn event with qmp command as:
{'execute': 'input-send-event', 'arguments': {'events': [{'type': 'btn', 'data': {'down': True, 'button': 'left'}}]}, 'id': 'W3uMGZHZ'}
send scroll event is same with send btn event, just only has True option, none False option, like:
{'execute': 'input-send-event', 'arguments': {'events': [{'type': 'btn', 'data': {'down': True, 'button': 'wheel-up'}}]}, 'id': 'Qohmxaei'}
{'execute': 'input-send-event', 'arguments': {'events': [{'type': 'btn', 'data': {'down': True, 'button': 'wheel-down'}}]}, 'id': 'VqEnRfs1'}
Due to they have same structure, and I want to separate the _btn_event() and _scroll_event away,  so I call self._btn_event() directly for _scroll_event(),  just I should not use "event" param, it should define scroll action(wheel-up or wheel-down),  "event" is not so clear here, I want change it to "scroll", do you think so that?

Reply:
Ok, thanks for the clarification! so I think we'd better 'leave a comment for that.

Btw, doesn't it need to be followed with an up event?

Reply:
updated the code, and yes, qemu doesn't need to do up action for wheel-up and wheel-down, only for btn(left, right, middle,side, extra) need do up action followed down action.